.gitlab-ci.yml

```
@@ -47,6 +50,42 @@ variables:
     paths:
       - ./artifacts/
 
+### stage:                        nightly-test
+
+nightly-test:
+  stage:                          nightly-test
+  <<:                             *docker-env
+  rules:
+    - if: $CI_PIPELINE_SOURCE == "schedule"
+  script:
+    - git clone -b master --depth 1 https://github.com/paritytech/polkadot.git polkadot
+    - cd polkadot
+    - cargo build --release --features fast-runtime
+    - mkdir -p ~/.local/bin
+    - mv ./target/release/polkadot /usr/bin
+    - polkadot --version
+    - cd -
+    - rm -rf polkadot
+    - RUST_LOG=info cargo test --workspace --all-features -- --nocapture
+    - cargo clean
+
+nightly-test-failure:
+  stage:                           nightly-test
+  image:                           "${CI_IMAGE}"
+  rules:
+    - if: $CI_PIPELINE_SOURCE == "schedule"
+  needs:                           ["nightly-test"]
+  when:                            on_failure
+  script:
+    - 'curl
+        -X POST
+        -H "Accept: application/vnd.github+json"
+        -H "Authorization: token ${GITHUB_TOKEN}"
+        "https://api.github.com/repos/paritytech/staking-miner-v2/actions/workflows/nightly-fail.yml/dispatches"
+        -d $"{\"ref\":\"main\",\"inputs\":{\"gitlab_pipeline\":\"${CI_PIPELINE_URL}\"}}"'
+  tags:
+    - linux-docker
```

```suggestion
    - kubernetes-parity-build
```
Maybe it's better to run it in k8s?

---

Agreed. Also changed an image against thinner `paritytech/tools:latest`